service/routes.py

```diff
@@ -66,6 +89,33 @@ def create_products():
 
 
 ######################################################################
+# UPDATE A PRODUCT
+######################################################################
+@app.route("/products", methods=["PUT"])
+def update_product():
+    """
+    Update a Product
+    This endpoint will update a existing Product based the data in the body that is posted
+    or return 404 there is no product with id provided in payload
+    """
+
+    app.logger.info("Request to update a product")
+    check_content_type("application/json")
+
+    product_id = request.get_json()["id"]
```

Comment:
we should all use rest api. get the product_id in the url not parse the json.

/products/<int:product_id>